packages/driver/src/util/serialization/log.ts

```diff
@@ -327,6 +327,13 @@ export const reifyLogLikeFromSerialization = (props, matchElementsAgainstSnapsho
  * @returns a serializable form of a snapshot, including a serializable <body> with styles
  */
 export const preprocessSnapshotForSerialization = (snapshot) => {
+  // if the protocol is enabled, we don't need to preprocess the snapshot since it is serializable,
+  // also make sure numTestsKeptInMemory is 0, otherwise we will want to preprocess the snapshot
+  // (the driver test's set numTestsKeptInMemory to 1 in run mode to verify the snapshots)
+  if (Cypress.config('protocolEnabled') && Cypress.config('numTestsKeptInMemory') === 0) {
```

Comment:
OK so in other words this is handled elsewhere by protocol, and if we are in run mode with (or I guess open as well) with `numTestsKeptInMemory=0` and `protocolEnabled` is true, this is a no-op?

I wonder if this should return null instead because those logs still get serialized, right? What happens right now if we send the snapshot as is through postmessage?

Reply:
Protocol is only enabled in run mode. We want the [partial snapshot](https://github.com/cypress-io/cypress/blob/feat/protocol/packages/driver/src/cy/snapshots.ts#L253) to get serialized so it gets saved to the database.